torch/optim/adam.py

```diff
@@ -174,9 +175,14 @@ def _init_group(
                         else torch.tensor(0.0, dtype=_get_scalar_dtype())
                     )
                     # Exponential moving average of gradient values
-                    state["exp_avg"] = torch.zeros_like(
-                        p, memory_format=torch.preserve_format
+                    # case beta1 == 0, we don't need exp_avg
+
+                    state["exp_avg"] = (
+                        torch.zeros_like(p, memory_format=torch.preserve_format)
+                        if beta1 > 0
+                        else torch.zeros(0)
```

Comment:
```suggestion
                        else None
```
Wait why do we have this entry at all if beta1 is 0? We should make exp_avg optionally exist in the implementations.